django_redis/cache.py

```diff
@@ -1,17 +1,21 @@
 import functools
 import logging
+import time
+from threading import RLock
 
 from django.conf import settings
 from django.core.cache.backends.base import BaseCache
 
-from .util import load_class
 from .exceptions import ConnectionInterrupted
-
+from .util import load_class
 
 DJANGO_REDIS_IGNORE_EXCEPTIONS = getattr(settings, "DJANGO_REDIS_IGNORE_EXCEPTIONS", False)
 DJANGO_REDIS_LOG_IGNORED_EXCEPTIONS = getattr(settings, "DJANGO_REDIS_LOG_IGNORED_EXCEPTIONS", False)
 DJANGO_REDIS_LOGGER = getattr(settings, "DJANGO_REDIS_LOGGER", False)
 DJANGO_REDIS_SCAN_ITERSIZE = getattr(settings, "DJANGO_REDIS_SCAN_ITERSIZE", 10)
+DJANGO_REDIS_EXCEPTION_THRESHOLD = getattr(settings, "DJANGO_REDIS_EXCEPTION_THRESHOLD", None)
+DJANGO_REDIS_EXCEPTION_THRESHOLD_TIME_WINDOW = getattr(settings, "DJANGO_REDIS_EXCEPTION_TIME_WINDOW", 1)
+DJANGO_REDIS_EXCEPTION_THRESHOLD_COOLDOWN = getattr(settings, "DJANGO_REDIS_EXCEPTION_THRESHOLD_COOLDOWN", 5)
```

Comment:
An alternative way might be a "maximum total exception timeout" and a "cooldown time" where you can set 2 times, say '15 seconds',  '5 minutes' (respectively) and if the total time spent on timeouts on back to back requests without a success is 15 seconds, silence exceptions for 5 minutes